### PR TITLE
[FIX] [16.0] contract: Show contract button on partner when they have contracts too

### PR DESCRIPTION
**Pull request by Shide**
Show button on partner when they have contracts

fixes https://github.com/OCA/contract/issues/1143

MT-7885 @moduon @rafaelbn @Gelojr @fcvalgar @rousseldenis please review if you want :)

### CODE REVIEW

File: contract/views/res_partner_view.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
                     groups="account.group_account_invoice"
                     icon="fa-book"
                     context="{'default_contract_type': 'sale', 'contract_type': 'sale'}"
-                    attrs="{'invisible': [('customer_rank','=',0)]}"
+                    attrs="{'invisible': [('sale_contract_count','=',0)]}"
```
Comment (carlosdauden) [inline contract/views/res_partner_view.xml]: With this change the contracts button will not appear when creating a new customer.

When creating a customer from, for example, Invoicing > Customers the customer created is already assigned a customer_rank value, so I don't understand the problem with the current behaviour.

https://github.com/odoo/odoo/blob/62af1c7b04557e94d18d6078544dc867e61ef2d2/addons/account/views/partner_view.xml#L280

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: contract/views/res_partner_view.xml (hunk 1 above)
Comment (fcvalgar) [reply contract/views/res_partner_view.xml]: In our use case, we use the Members module, and contacts are created from this module. Thank you for the comment!

**Review by fcvalgar — APPROVED** [review]
Functional review.

LGTM, Thank you @Shide.

image

image

File: contract/views/res_partner_view.xml (hunk 1 above)
Comment (rousseldenis) [reply contract/views/res_partner_view.xml]: I would say, this is semantically more correct as the context is contract oriented. So, having contracts to show the button is better than 'being a customer'.

BUT, I'm wondering if showing it even in no contracts is better or not as with that feature, user can create a new contract for the partner directly. What do you think of ?

Comment (carlosdauden) [inline contract/views/res_partner_view.xml]: Let's see if whith this we can find a consensus solution:
```suggestion
                    attrs="{'invisible': [('customer_rank','=',0), ('sale_contract_count','=',0)]}"
```

File: contract/views/res_partner_view.xml [hunk 2]
```diff
@@ -40,7 +40,7 @@
                     groups="account.group_account_invoice"
                     icon="fa-book"
                     context="{'default_contract_type': 'purchase', 'contract_type': 'purchase'}"
-                    attrs="{'invisible': [('supplier_rank','=',0)]}"
+                    attrs="{'invisible': [('purchase_contract_count','=',0)]}"
```
Comment (carlosdauden) [inline contract/views/res_partner_view.xml]:
```suggestion
                    attrs="{'invisible': [('supplier_rank','=',0), ('purchase_contract_count','=',0)]}"
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: contract/views/res_partner_view.xml (hunk 1 above)
Comment (rousseldenis) [reply contract/views/res_partner_view.xml]: Ouch, I would say this is worse...

**Comment by Shide** [comment]
@carlosdauden sure, your reasoning makes sense. 
Applied both conditions to invisible attr so the default values can be applied directly on the first contract.

File: contract/views/res_partner_view.xml (hunk 1 above)
Comment (carlosdauden) [reply contract/views/res_partner_view.xml]: A fairly common flow is to create the customer from the Invoicing > Customers option and once created click the smartbutton to create a new contract.
Any changes that involve losing the most basic flow to solve data integrity problems or "incorrect" processes by the user should not be included in the "base" modules.

**Review by carlosdauden — APPROVED** [review]

File: contract/views/res_partner_view.xml (hunk 1 above)
Comment (rousseldenis) [reply contract/views/res_partner_view.xml]: > A fairly common flow is to create the customer from the Invoicing > Customers option and once created click the smartbutton to create a new contract.

customer_rank == 0 if you create it from `Contacts` menu. So, this is not allowed in that case. Removing at all the domain is fair enough IMHO.

**Review by rafaelbn — APPROVED** [review]
/ocabot merge minor

**Pull request merged**